### DISCUSSION TITLE
How to solve [project.assets.json' doesn't have a target for 'net8.0/win-x64']

### DISCUSSION THREAD

**Discussion by RIldefonso** [unoplatform/uno#15990] [General]
I have a WAsm app to test UNO in VS 2022 very up to date, and I tried to publish it, but it has this error.
I had tried to restore, rebuild, reinit...

Again, I can't do anything with UNO.
How to solve?

**Comment by patlevin** [comment]
I ran into the same issue and found that adding 

```xml
<PropertyGroup>
    ...
    <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
</PropertyGroup>
```

to the wasm project fixed the issue.

However, I then immediately ran into another problem: publish didn't succeed (neither from within Visual Studio, nor using the `dotnet publish` directly from the command line), because msbuild complained that some process was locking the output assembly.

After some searching and analyzing the build log, I managed to get one step further by passing the `-m:1` parameter to `dotnet publish`, forcing `msbuild` to only use a single CPU core. This solved the locked file issue, but now I'm stuck at an error saying

> WasmAssembliesToBundle item is empty. No assemblies to process.

I'll let you know if I find out how to solve that. Sigh.

**Reply by jeromelaban** [reply]
Thanks for the reports. This is not a behavior we've seen, and it would help if you were to provide a repro. Also, [providing a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) generally helps further.

**Reply by patlevin** [reply]
The problem seems to be a completely broken solution file. I was unable to get the original project working with wasm at all.

In the end I created a new project using the *Blank* startup type, XAML markup, no presentation framework package, default WinUI Frame Navigation, no VSCode debugging, and no authentication.

I copied my original code files over to the newly created project, added a single missing NuGet dependency and tried to build and publish the wasm configuration. This time everything worked fine with the new project.

This leads me to believe that one of the NuGet dependencies I in the broken project was causing the problem. Due to time constraints I am currently unable to narrow down, which package in particular is causing the issue.

**Reply by wiz0u** [reply]
> ```
> <PropertyGroup>
>     ...
>     <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
> </PropertyGroup>
> ```

For me, I had to use `<RuntimeIdentifiers>win-x64</RuntimeIdentifiers>` (**plural, with an s**) to solve my publish problem

**Comment by AdriaanLarcai** [comment]
I've just run into this issue as well on the latest stable release of Uno.Sdk (5.5.43).

Brand new Uno Library project targeting dotnet 9 WinUI and WASM. See below:
```
<Project Sdk="Uno.Sdk">
    <PropertyGroup>
        <TargetFrameworks>net9.0-windows10.0.26100;net9.0-browserwasm</TargetFrameworks>
        <UnoSingleProject>true</UnoSingleProject>
        <OutputType>Library</OutputType>
        <!-- Ensures the .xr.xml files are generated in a proper layout folder -->
        <GenerateLibraryLayout>true</GenerateLibraryLayout>
        <ImplicitUsings>disable</ImplicitUsings>
        <Nullable>disable</Nullable>
        <DefaultLanguage>en-US</DefaultLanguage>
        <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
        <!--
      UnoFeatures let's you quickly add and manage implicit package references based on the features you want to use.
      https://aka.platform.uno/singleproject-features
    -->
        <UnoFeatures>
            Hosting;
            Logging;
            Mvvm;
        </UnoFeatures>
        <Platforms>AnyCPU;x64</Platforms>
    </PropertyGroup>
</Project>
```

Running into the same exception:

```C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Sdk\9.0.0\Sdk\WasmApp.Common.targets(365,5): error : WasmAssembliesToBundle item is empty. No assemblies to process```

This appears to be a non-issue with the latest dev release at the time of writing: `Uno.Sdk 5.6.0-dev.192`
So I'm hoping that it will remain fixed for the stable release of 5.6.

**Reply by jeromelaban** [reply]
Thanks for the report, see https://github.com/unoplatform/uno/issues/18860 for a workaround.
